J - news-summary-challenge

J-son1 opened the pull request.

J-son1 committed:
First commit

J-son1 committed:
Set up project

J-son1 committed:
Add Articles model that can return articles

J-son1 committed:
Add reset method to Articles model

J-son1 committed:
Add ArticlesView class with displayArticles method

J-son1 committed:
Modify main file to display article headlines on the page

J-son1 committed:
Add an API class with loadArticles method to fetch articles from The Guardian API

J-son1 committed:
Update the view class display method to handle api data

J-son1 committed:
Update the main file to display headlines from The Guardian API

J-son1 committed:
Update the view class so that an image is displayed with each article headline

J-son1 committed:
Modify existing tests for the view class

J-son1 committed:
Link each headline to their original article

J-son1 committed:
Add a search field to allow the user to see articles that match the search query

fpmrqs commented on articlesView.test.js:
```diff
@@ -0,0 +1,77 @@
+const articlesData = [
+  {headline: "First article", thumbnail: "./images/image1.jpeg", webUrl: "https://www.google.com/"},
+  {headline: "Second article", thumbnail: "./images/image2.jpeg", webUrl: "https://www.google.com/"}, 
+  {headline: "Third article", thumbnail: "./images/image3.jpeg", webUrl: "https://www.google.com/"}
```
great mocking skills, 10/10

fpmrqs commented on articlesView.js:
```diff
@@ -0,0 +1,60 @@
+    });
+  }
+
+  displayArticles() {
```
Creating a few private functions to break this function down would help with the SRP

J-son1 replied:
Thank you! This is really good advice and I noticed you did this very well in your project which was really great to see.

fpmrqs commented on articlesModel.js:
```diff
@@ -0,0 +1,19 @@
+    return this.articles;
+  }
+
+  reset() {
```
The reset function might not be necessary, as the setArticles function overrides this.articles? I'm not entirely sure though lol I could be very wrong.

fpmrqs commented on articlesApi.js:
```diff
@@ -0,0 +1,25 @@
+      });
+  }
+
+  #queryToUrl(queryStr) {
```
I like how you implemented the query, looks smart

fpmrqs reviewed (commented):
It was extra hard to give meaningful feedback as I feel we had very similar approaches. Very good testing, looks like the coverage is high. I actually learned a few things about how to improve my tests, so thank you for that :) Hopefully the suggestions in the comments will be useful to you. Well done!

J-son1 committed:
Amend tests by mocking classes

J-son1 committed:
Add article link to image

J-son1 committed:
Add styling

J-son1 committed:
Fix tests

J-son1 committed:
Refactor articlesApi

J-son1 committed:
Modify page heading styles

J-son1 committed:
Fix search feature, tests, styling. Refactor

- make search feature work with single and multiple word input
- fix broken tests
- amend styling for search bar
- refactor for readablity

J-son1 committed:
Amend AritclesApi and search feature

J-son1 committed:
Update README

J-son1 committed:
Centre image on README

J-son1 committed:
Modify README

J-son1 committed:
Amend README

J-son1 committed:
Modify images
